lib/runtime/wasmer/instance.go

```diff
@@ -156,25 +156,24 @@ func (in *Instance) UpdateRuntimeCode(code []byte) (err error) {
 	return nil
 }
 
-// CheckRuntimeVersion calculates runtime Version for runtime blob passed in
-func (in *Instance) CheckRuntimeVersion(code []byte) (runtime.Version, error) {
-	in.mutex.Lock()
-	defer in.mutex.Unlock()
-
-	wasmInstance, allocator, err := setupVM(code)
+// CheckRuntimeVersion finds the runtime version by initiating a temporary
+// runtime instance using the WASM code provided, and querying it.
+func CheckRuntimeVersion(code []byte) (version runtime.Version, err error) {
```

Comment:
like @timwu20 mentioned there isn't really a need for this function to remain in the runtime/wasmer package anymore, might make more sense for it to be in dot/state

Reply:
Kept it there to avoid replicating 3 times (lib/runtime/wasmer, dot/state, and somewhere else too) as per what Tim described.

Also it might get a bit more complex due to `Core_version` being deprecated, and we won't need to create an instance anymore for newer runtimes, so we should definitely have it in one place. (see https://github.com/paritytech/substrate/pull/8688)